client/my-sites/plans-features-main/hooks/use-eligibility-for-term-savings-price-display.ts

```diff
@@ -2,32 +2,52 @@ import { isEnabled } from '@automattic/calypso-config';
 import {
 	getPlanSlugForTermVariant,
 	PlanSlug,
+	TERMS_LIST,
 	URL_FRIENDLY_TERMS_MAPPING,
 	UrlFriendlyTermType,
 } from '@automattic/calypso-products';
 import { AddOns, Plans } from '@automattic/data-stores';
-import { GridPlan } from '@automattic/plans-grid-next';
+import { HiddenPlans, PlansIntent, usePlanTypesWithIntent } from '@automattic/plans-grid-next';
+import usePlansFromTypes from '@automattic/plans-grid-next/src/hooks/data-store/use-plans-from-types';
 import useLongerPlanTermDefaultExperiment from './experiments/use-longer-plan-term-default-experiment';
 import useCheckPlanAvailabilityForPurchase from './use-check-plan-availability-for-purchase';
 
 const useEligibilityForTermSavingsPriceDisplay = ( {
-	gridPlans,
+	hiddenPlans,
+	intent,
+	isSubdomainNotGenerated,
+	selectedPlan,
+	term,
 	displayedIntervals,
 	coupon,
 	siteId,
 	storageAddOns,
 	isInSignup,
 }: {
-	gridPlans: GridPlan[];
+	hiddenPlans?: HiddenPlans;
+	intent?: PlansIntent;
+	isSubdomainNotGenerated?: boolean;
+	selectedPlan?: PlanSlug;
+	term: ( typeof TERMS_LIST )[ number ];
 	displayedIntervals: UrlFriendlyTermType[];
 	storageAddOns: ( AddOns.AddOnMeta | null )[] | null;
 	coupon?: string;
 	siteId?: number | null;
 	isInSignup?: boolean;
 } ) => {
 	const longerPlanTermDefaultExperiment = useLongerPlanTermDefaultExperiment();
-	const planSlugs = gridPlans.map( ( { planSlug } ) => planSlug );
```

Comment:
p.s. NIT - if we bring back the original (prior to these changes), would it make sense to pass `planSlugs` instead? So do the mapping at consuming end perhaps.

Reply:
If we go back to the original, I prefer `planSlugs`. It more concisely describes what's needed for the hook, and it generalizes the interface more.